linked-lists/c-or-cpp/josephus-problem.c

```diff
@@ -0,0 +1,63 @@
+/*Circular linked list implementation in C Programming
+has been used to solve this problem*/
+#include <stdio.h>
+#include <stdlib.h>
+struct Node{
+    int data;
+    struct Node *next;
+}*first,*last;
+void create (int a[],int n){// for creating nodes
+    struct Node *t;
+    first=(struct Node*)malloc(sizeof(struct Node));
+    first->data=a[0];
+    first->next=NULL;
+    last=first;
+    for(int i=1;i<n;i++){
+        t=(struct Node*)malloc(sizeof(struct Node));
+        t->data=a[i];
+        t->next=NULL;
+        last->next=t;
+        last=t;
+    }
+    last->next=first;
+
+}
+void display(struct Node *p){ //displaying nodes
+ printf("The Linked List is successfully created\n");
+   while(p->next!=first){
+       printf("->%d",p->data);
+       p=p->next;
+   }
+   printf("->%d",p->data);
+}
+void josephus(int k, struct Node *first){ //josephus implementation
+    struct Node *trail;
+    while(first->next!=first){
```

Comment:
In case value of `k` is 1 
the following will lead to undefined behaviour as the value of `trail` is not initialised before hand. 
the result might end up as an infinite loop or segmentation fault.